netket/experimental/operator/_fermions_2nd.py

```diff
@@ -0,0 +1,605 @@
+
+                    empty_site = is_empty(xt[orb_idx])
+
+                    if dagger:
+                        if not empty_site:
+                            has_xp = False
+                        else:
+                            mel *= (-1) ** np.sum(xt[:orb_idx])  # jordan wigner sign
+                            xt[orb_idx] = flip(xt[orb_idx])
+                    else:
+                        if empty_site:
+                            has_xp = False
+                        else:
+                            mel *= (-1) ** np.sum(xt[:orb_idx])  # jordan wigner sign
+                            xt[orb_idx] = flip(xt[orb_idx])
+
+                # if this is the end of the term, we collect things
+                if term_end:
+
+                    if has_xp:
+                        x_prime[n_c, :] = xt
+                        mels[n_c] = mel
+                        n_c += 1
+
+                    new_term = True
+
+            if pad:
+                n_c = (b + 1) * max_conn
+
+            sections[b] = n_c
+
+        if pad:
+            return x_prime, mels
+        else:
+            return x_prime[:n_c], mels[:n_c]
+
+    @staticmethod
+    def identity(hilbert):
+        """identity operator"""
+        return FermionOperator2nd(hilbert, [], [], constant=1.0)
+
+    @staticmethod
+    def zero(hilbert):
+        """returns an object that has no contribution, meaning a constant of 0"""
+        return FermionOperator2nd(hilbert, [], [], constant=0.0)
+
+    def _op__matmul__(self, other):
+        if not isinstance(other, FermionOperator2nd):
+            return NotImplementedError
+        if not self.hilbert == other.hilbert:
+            raise ValueError(
+                f"Can only multiply identical hilbert spaces (got A@B, A={self.hilbert}, B={other.hilbert})"
+            )
+        terms = []
+        weights = []
+        for t, w in zip(self._orig_terms, self._orig_weights):
+            for to, wo in zip(other._orig_terms, other._orig_weights):
+                terms.append(tuple(t) + tuple(to))
+                weights.append(w * wo)
+        if other._constant != 0.0:
+            for t, w in zip(self._orig_terms, self._orig_weights):
+                terms.append(tuple(t))
+                weights.append(w * other._constant)
+        if self._constant != 0:
+            for t, w in zip(other._orig_terms, other._orig_weights):
+                terms.append(tuple(t))
+                weights.append(w * self._constant)
+        constant = self._constant * other._constant
+        dtype = np.promote_types(self.dtype, other.dtype)
+        return FermionOperator2nd(
+            self.hilbert, terms, weights=weights, constant=constant, dtype=dtype
+        )
+
+    def __radd__(self, other):
+        return self + other
+
+    def __add__(self, other):
+        if np.issubdtype(type(other), np.number):
+            if other != 0.0:
+                return self + FermionOperator2nd.identity(self.hilbert) * other
```

Comment:
ah i see now. I'd still put this in `nkx.operator.fermion.identity`. If you want you can leave the identity function in this file but make it internal, and possibly not a class method.

The reason is that the code will be easier to refactor in the future.